Helper/SSOHelper.php

```diff
@@ -17,11 +17,111 @@
 
 namespace Bolt\Boltpay\Helper;
 
+use Bolt\Boltpay\Helper\Api as ApiHelper;
+use Bolt\Boltpay\Helper\Config as ConfigHelper;
+use Exception;
+use Magento\Framework\App\Helper\AbstractHelper;
+use Magento\Framework\App\Helper\Context;
+use Magento\Framework\DataObjectFactory;
+use Magento\Store\Model\StoreManagerInterface;
+
 /**
  * Helper for SSO
  */
-class SSOHelper
+class SSOHelper extends AbstractHelper
 {
+    /**
+     * @var ConfigHelper
+     */
+    private $configHelper;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @var DataObjectFactory
+     */
+    private $dataObjectFactory;
+
+    /**
+     * @var ApiHelper
+     */
+    private $apiHelper;
+
+    /**
+     * @param Context               $context
+     * @param ConfigHelper          $configHelper
+     * @param StoreManagerInterface $storeManager
+     * @param DataObjectFactory     $dataObjectFactory
+     * @param ApiHelper             $apiHelper
+     */
+    public function __construct(
+        Context $context,
+        ConfigHelper $configHelper,
+        StoreManagerInterface $storeManager,
+        DataObjectFactory $dataObjectFactory,
+        ApiHelper $apiHelper
+    ) {
+        parent::__construct($context);
+        $this->configHelper = $configHelper;
+        $this->storeManager = $storeManager;
+        $this->dataObjectFactory = $dataObjectFactory;
+        $this->apiHelper = $apiHelper;
+    }
+
+    /**
+     * Return information needed for oauth exchange
+     *
+     * @return array
+     */
+    public function getOAuthConfiguration()
+    {
+        $storeId = $this->storeManager->getStore()->getId();
+
+        $publishableKey = $this->configHelper->getPublishableKeyCheckout($storeId);
+        $publishableKeySplit = explode('.', $publishableKey);
+        $clientID = end($publishableKeySplit);
```

Comment:
comment here plz

Reply:
for?

Reply:
indicating why the last piece is the client ID

Reply:
i'll update the doc for the function to make it more clear what each of these thigns are

Reply:
updated